### PR TITLE
[TA2160]Adding test case in controller for testing isDeleteEvent function.

### PR DESCRIPTION
**Pull request by sonasingh46**
This commit will add test cases for IsDeleteEvent
function in controller.

Signed-off-by: sonasingh46 <sonasingh46@gmail.com>

### CODE REVIEW

File: cmd/maya-apiserver/spc-watcher/controller_test.go [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+import (
+	"testing"
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	api_meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"time"
+)
+
+func TestIsDeleteEvent(t *testing.T) {
+	// SPC object with deletion timestamp set to nil
+	fakespcobject1 := &apis.StoragePoolClaim{}
```
Comment (princerachit) [inline cmd/maya-apiserver/spc-watcher/controller_test.go]: Can these be put directly inside the `tests` map?

File: cmd/maya-apiserver/spc-watcher/controller_test.go [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+import (
+	"testing"
+	apis "github.com/openebs/maya/pkg/apis/openebs.io/v1alpha1"
+	api_meta "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"time"
+)
+
+func TestIsDeleteEvent(t *testing.T) {
+
```
Comment (princerachit) [inline cmd/maya-apiserver/spc-watcher/controller_test.go]: Please remove empty lines.

File: cmd/maya-apiserver/spc-watcher/controller_test.go [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+	tests := map[string]struct {
+		fakestoragepoolclaim *apis.StoragePoolClaim
+		expectedResult       bool
+	}{
+		"Setting DeletionTimestamp to nil and it should return false": {
```
Comment (princerachit) [inline cmd/maya-apiserver/spc-watcher/controller_test.go]: The test message could be improved to `Unset deletion timestamp`

File: cmd/maya-apiserver/spc-watcher/controller_test.go [hunk 4]
```diff
@@ -0,0 +1,55 @@
...
+			fakestoragepoolclaim: &apis.StoragePoolClaim{
+				ObjectMeta: api_meta.ObjectMeta{
+					DeletionTimestamp: nil,
+				},
+			},
+			expectedResult: false},
+		"Setting DeletionTimestamp not nil and it should return true": {
```
Comment (princerachit) [inline cmd/maya-apiserver/spc-watcher/controller_test.go]: This test message could be improved to `Set/Present deletion timestamp`

File: cmd/maya-apiserver/spc-watcher/controller_test.go (hunk 3 above)
Comment (princerachit) [reply cmd/maya-apiserver/spc-watcher/controller_test.go]: You may add positive/negative as prefix

File: cmd/maya-apiserver/spc-watcher/controller_test.go (hunk 2 above)
Comment (sonasingh46) [reply cmd/maya-apiserver/spc-watcher/controller_test.go]: This is for readability

**Pull request merged**